Frontend/src/Base/BaseImageUpload/BaseImageUpload.jsx

```diff
@@ -0,0 +1,111 @@
+import { useRef, useState } from 'react';
+import BaseButton from '../BaseButton';
+import { useTranslation } from 'react-i18next';
+import UploadedImagesDisplay from './UploadedImagesDisplay';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectAddPlace, addPlaceActions } from '../../Redux/addPlaceSlice';
+
+const BaseImageUpload = ({ fileSize }) => {
+  const dispatch = useDispatch();
+  const images = useSelector(selectAddPlace).images;
+  const [errorMsg, setErrorMsg] = useState(null);
+  const [showError, setShowError] = useState(null);
+  const fileInputRef = useRef(null);
+  const { t } = useTranslation();
+  const MAX_FILE_SIZE = fileSize * 1024 * 1024;
+  const MAX_FILES = 3;
+
+  const handleClick = () => {
+    fileInputRef.current.click();
+  };
+
+  const validateImagesQuantity = (uploadedImagesLength) => {
+    const allFilesLenght = uploadedImagesLength + images.length;
+
+    if (uploadedImagesLength < MAX_FILES || allFilesLenght < MAX_FILES) {
+      return true;
+    } else {
```

Comment:
Miko to bez sensu jest ten else jaka jest inna opcja poza tym ifem w fukcji ?
```
 const validateDuplicates = (uploadedImage) => {
    const duplicateFile = images.find((image) => image.name === uploadedImage.name);
    if (!duplicateFile) {
      return true;
    } 
    setErrorMsg(`${t('common.already_uploaded')} ${uploadedImage.name}`);
    return false;
  };

Reply:
Uda się if zwróci true nie uda się to setError nie ma potrzeby używać else i polecam ci tego nie robić

Reply:
Daro za dużo zbieżnych informacji, poprzednie przykłady wyglądały inaczej, już rozumiem o co ci chodzi, poprawię jak wrócę z pracy.